src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +36,72 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * @experimental in version 3.3
+     */
+    public function registerClasses(Definition $prototype, $namespace, $directoryGlob)
+    {
+        if ('\\' !== substr($namespace, -1)) {
+            throw new InvalidArgumentException(sprintf('Namespace prefix must end with a "\\": %s.', $namespace));
+        }
+        if (!preg_match('/^(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+\\\\)++$/', $namespace)) {
+            throw new InvalidArgumentException(sprintf('Namespace is not a valid PSR-4 prefix: %s.', $namespace));
+        }
+
+        if (strlen($directoryGlob) === $i = strcspn($directoryGlob, '*?{[')) {
+            $directoryPrefix = $directoryGlob;
+            $directoryGlob = '';
+        } else {
+            $directoryPrefix = dirname(substr($directoryGlob, 0, 1 + $i));
+            $directoryGlob = substr($directoryGlob, strlen($directoryPrefix));
+        }
+
+        $directoryPrefix = $this->locator->locate($directoryPrefix, $this->currentDir, true);
+        $directoryPrefix = realpath($directoryPrefix) ?: $directoryPrefix;
+        $directoryGlob = $directoryPrefix.$directoryGlob;
+
+        // track directories only for new & removed files
+        $this->container->fileExists($directoryPrefix, '/^$/');
+
+        $classes = $this->findClasses($namespace, $directoryGlob, strlen($directoryPrefix));
+
+        foreach ($classes as $class) {
+            $this->container->setDefinition($class, clone $prototype);
+        }
+    }
+
+    private function findClasses($namespace, $directoryGlob, $prefixLen)
+    {
+        $classes = array();
+        $extRegexp = defined('HHVM_VERSION') ? '/\\.(?:php|hh)$/' : '/\\.php$/';
+
+        foreach (glob($directoryGlob, defined('GLOB_BRACE') ? GLOB_BRACE : 0) as $path) {
```

Comment:
The conditional here means that the available features depend on the availability of ``GLOB_BRACE``, which may make the code less portable.
In which cases can it be undefined ?